examples/cpp/CGroupTest.cc

```diff
@@ -0,0 +1,79 @@
+/*
+ * CPUDistribution Show load distribution across CPU cores during a period of
+ *                 time. For Linux, uses BCC, eBPF. Embedded C.
+ *
+ * Basic example of BCC and kprobes.
+ *
+ * USAGE: CPUDistribution [duration]
+ *
+ * Copyright (c) Facebook, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ */
+
+#include <unistd.h>
+#include <fstream>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+#include "BPF.h"
+
+const std::string BPF_PROGRAM = R"(
+#include <linux/sched.h>
+#include <linux/path.h>
+#include <linux/dcache.h>
+
+BPF_CGROUP_ARRAY(cgroup, 1);
+
+int on_vfs_open(struct pt_regs *ctx, struct path *path) {
+  if (cgroup.check_current_task(0))
```

Comment:
This condition is not enough. The kernel implementation is below.
```
BPF_CALL_2(bpf_current_task_under_cgroup, struct bpf_map *, map, u32, idx)
{
        struct bpf_array *array = container_of(map, struct bpf_array, map);
        struct cgroup *cgrp;

        if (unlikely(idx >= array->map.max_entries))
                return -E2BIG;

        cgrp = READ_ONCE(array->ptrs[idx]);
        if (unlikely(!cgrp))
                return -EAGAIN;

        return task_under_cgroup_hierarchy(current, cgrp);
}
```
Initially if there is no cgroup is installed in the cgroup array,
-EAGAIN will be returned. We should not print the opened files
as it will be confusing as cgroup is not really tested.

I suggest the above condition is checked as:
```
  __s64 ret = cgroup.check_current_task(0);
 if (ret == 1) { ...} // if (ret > 0) { ...} should also work
```

Since this is a cgroup test, I think the cgroup path parameter should be required unless we want create a cgroup for user and mimic the test if user did not provide the path.

Please also include instructions how to setup the test environment as the comments in this file, so user does not need to find the commit messages for trying.

Reply:
Thank you so much for the detailed explanation, I will do that.